src/backend/renderer/utils/mod.rs

```diff
@@ -75,6 +75,15 @@ pub struct DamageSnapshot<N, Kind> {
     damage: Arc<VecDeque<smallvec::SmallVec<[Rectangle<N, Kind>; MAX_DAMAGE_RECTS]>>>,
 }
 
+impl<N, Kind> Clone for DamageBag<N, Kind> {
+    fn clone(&self) -> Self {
+        Self {
+            limit: self.limit,
+            state: self.state.clone(),
+        }
+    }
+}
```

Comment:
I intentionally did not implement `Clone` on `DamageBag`. `DamageBag::snapshot` should instead be used to get a read-only view of the current damage.